yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1183,6 +1187,29 @@ object Client extends Logging {
     YarnSparkHadoopUtil.addPathToEnvironment(env, Environment.CLASSPATH.name, path)
 
   /**
+   * Returns the path to be sent to the NM for a local path string.
+   *
+   * This method uses two configuration values:
+   *
+   * - spark.yarn.config.localPath: a string that identifies a portion of the input path that may
+   *   only be valid in the local process.
+   * - spark.yarn.config.clusterPath: a string with which to replace the local path. This may
+   *   contain, for example, env variable references, which will be expanded by the NMs when
+   *   starting containers.
+   *
+   * If either config is not available, the input path is returned.
+   */
+  def getClusterPath(conf: SparkConf, path: String): String = {
+    val localPath = conf.get("spark.yarn.config.localPath", null)
+    val clusterPath = conf.get("spark.yarn.config.clusterPath", null)
```

Comment:
this is pretty minor, but can we add a check that there are no special characters in this confs?  Eg., if it contains a ":" would be nice to give the user an exception rather than some weird behavior